MakeHist/mumu.C

```diff
@@ -0,0 +1,337 @@
+                       lepphi_hist->Fill(leptons[0].Phi());
+                       lepphi_hist->Fill(leptons[1].Phi());
+
+                       //Iso
+                       lepIso_hist->Fill(leptons_[0].Iso);
+                       lepIso_hist->Fill(leptons_[1].Iso);
+
+                       //lep-lep dPhi and dR, dilep mass
+                       dPhi_hist->Fill(Dil_dPhi[0]);
+                       dR_hist->Fill(Dil_dR[0]);
+                       mass_hist->Fill(Dil_mass[0]);
+
+                       //Jet1_nonB: Non-bjet with the largest pT
+                       //Jet2_nonB: Non-bjet with the second largest pT
+                       int Jet1_nonB=-1;
+                       int Jet2_nonB=-1;
+                       float Jet1_nonB_PT=0.;
+                       float Jet2_nonB_PT=0.;
+
+                       //Jet1_B: b-jet with the largest pT
+                       //Jet2_B: b-jet with the second largest pT
+                       int Jet1_B=-1;
+                       int Jet2_B=-1;
+                       float Jet1_B_PT=0.;
+                       float Jet2_B_PT=0.;
+                       float All_bjet_PT=0.;
+
+                       if(Nbjet == 0){
+                           for(unsigned Nj=0; Nj<Jets.size(); ++Nj){
+                               if(Jet1_nonB_PT<Jets[Nj].Pt()){
+                                   Jet1_nonB=Nj;
+                                   Jet1_nonB_PT=Jets[Nj].Pt();
+                               }
+                               if(Jet2_nonB_PT < Jets[Nj].Pt() && Jet1_nonB_PT > Jets[Nj].Pt()){
+                                   Jet2_nonB = Nj;
+                                   Jet2_nonB_PT=Jets[Nj].Pt();
+                               }
+                           }
+                       }
+
+                       else if(Nbjet == 1){
+                           for(unsigned Nj=0; Nj<Jets.size(); ++Nj){
+                               if(Jet2_nonB_PT<Jets[Nj].Pt()){
+                                   if(Jet2_nonB_PT != bJets[0].Pt()){
+                                       Jet2_nonB=Nj;
+                                       Jet2_nonB_PT=jet_pt[Nj];
+                                   }
+                               }
+                           }
+                       }
+                       else if(Nbjet > 1){
+                           for(unsigned Nbj=0; Nbj<bJets.size(); ++Nbj){
+                               All_bjet_PT += bJets[Nbj].Pt();
+                               if(Jet1_B_PT < bJets[Nbj].Pt()){
+                                   Jet1_B=Nbj;
+                                   Jet1_B_PT=bJets[Nbj].Pt();
+                               }
+                               if(Jet2_B_PT < bJets[Nbj].Pt() && Jet1_B_PT > bJets[Nbj].Pt()){
+                                   Jet2_B = Nbj;
+                                   Jet2_B_PT = bJets[Nbj].Pt();
+                               }
+                           }
+                       }
```

Comment:
This rather complicated section should simplify significantly for having two exclusive b-tagged and non-b-tagged jet collections. All the checks for differences in pT will then become superfluous.